test/parallel/test-promise-unhandled-warn-to-error.js

```diff
@@ -0,0 +1,28 @@
+// Flags: --expose-gc
+'use strict';
+
+const common = require('../common');
+
+process.env.NODE_UNHANDLED_REJECTION = 'SILENT';
+
+// Verify that setting the environment variable in a handler will this change
+// the behavior properly.
+
+Promise.reject('test');
+
+new Promise(() => {
+  throw new Error('One');
+});
+
+global.gc();
+
+process.on('unhandledRejection', () => {
+  process.env.NODE_UNHANDLED_REJECTION = 'ERROR';
```

Comment:
I'm not sure we should specify or support this but no strong feelings on it

Reply:
I was not sure either and first I did not implement it this way. This way we are just safe to always do exactly what a user expects. If we do not want to support this, we should likely not use an environment variable but I believe many users prefer those over a flag.